x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.ts

```diff
@@ -9,26 +9,65 @@ import path from 'path';
 import puppeteer from 'puppeteer-core';
 import rimraf from 'rimraf';
 import * as Rx from 'rxjs';
-import { map, share, mergeMap, filter, partition } from 'rxjs/operators';
+import { filter, map, mergeMap, partition, share } from 'rxjs/operators';
+import { safeChildProcess } from '../../safe_child_process';
 import { HeadlessChromiumDriver } from '../driver';
 import { args } from './args';
-import { safeChildProcess } from '../../safe_child_process';
 
-const compactWhitespace = (str) => {
+const compactWhitespace = (str: string) => {
   return str.replace(/\s+/, ' ');
 };
 
+interface IViewport {
+  width: number;
+  height: number;
+}
+
+interface IBrowserConfig {
+  disableSandbox: boolean;
+  proxy: any;
+}
+
+interface ILaunchArgs {
+  viewport: IViewport;
+  browserTimezone: string;
+}
+
 export class HeadlessChromiumDriverFactory {
-  constructor(binaryPath, logger, browserConfig) {
+  public type = 'chromium';
+  private binaryPath: string;
+  private logger: any;
+  private browserConfig: IBrowserConfig;
+
+  constructor(binaryPath: string, logger: any, browserConfig: IBrowserConfig) {
     this.binaryPath = binaryPath;
     this.logger = logger.clone(['chromium-driver-factory']);
     this.browserConfig = browserConfig;
   }
 
```

Comment:
It was a lot harder to use the `create` method here to do a simple smoke check, so I wrote a smaller `test` method to do so.